File: backend/models/photo.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel


class PhotoInDB(BaseModel):
    id: UUID
    user_id: UUID
    storage_path: str
    thumbnail_path: str | None = None
    original_filename: str | None = None
    file_size: int | None = None
    width: int | None = None
    height: int | None = None
    exif_data: dict[str, Any] = {}
    ai_tags: list[str] = []
    is_favorite: bool = False
    deleted_at: datetime | None = None
    created_at: datetime

File: mobile/lib/features/camera/domain/entities/photo.dart
class Photo {
  final String id;
  final String userId;
  final String storagePath;
  final String? thumbnailPath;
  final String? originalFilename;
  final int? fileSize;
  final int? width;
  final int? height;
  final Map<String, dynamic>? exifData;
  final List<String>? aiTags;
  final bool isFavorite;
  final DateTime? deletedAt;
  final DateTime createdAt;

  const Photo({
    required this.id,
    required this.userId,
    required this.storagePath,
    this.thumbnailPath,
    this.originalFilename,
    this.fileSize,
    this.width,
    this.height,
    this.exifData,
    this.aiTags,
    this.isFavorite = false,
    this.deletedAt,
    required this.createdAt,
  });
}
